packages/react/checkbox/src/Checkbox.tsx

```diff
@@ -57,12 +57,26 @@ const Checkbox = React.forwardRef<CheckboxElement, CheckboxProps>(
     const hasConsumerStoppedPropagationRef = React.useRef(false);
     // We set this to true by default so that events bubble to forms without JS (SSR)
     const isFormControl = button ? Boolean(button.closest('form')) : true;
+
+    const initialCheckState = React.useRef(defaultChecked ? true : checkedProp);
```

Comment:
It might make more sense to put that below `useControllableState` and use `checked` in the ref instead as this already should take care of the `defaultChecked` vs `checkedProp` case.

Also, then we can group it with `resetCheck` below.

Reply:
totally!